### PR TITLE
[WIP] symfony 3 support

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
This PR should enable symfony 3 support.

Todos:

* [ ] remove all form type text declarations, even in third pary (burgov)
* [ ] `_method` requirement removed
* [ ] remove setDefaultOptions()
* [ ] 'hidden' string removed
* [ ] remove `getRequest` call in Controller
* [ ] many more ...

### CODE REVIEW

File: Admin/Extension/SeoContentAdminExtension.php [hunk 1]
```diff
@@ -46,7 +47,7 @@ public function configureFormFields(FormMapper $formMapper)
             ->with($this->formGroup, array(
                 'translation_domain' => 'CmfSeoBundle',
             ))
-                ->add('seoMetadata', $isSf28 ? 'Symfony\Cmf\Bundle\SeoBundle\Form\Type\SeoMetadataType' : 'seo_metadata', array('label' => false))
```
Comment (WouterJ) [inline Admin/Extension/SeoContentAdminExtension.php]: `$isSf28` variable can now be removed

**Comment by ElectricMaxxx** [comment]
@WouterJ is there a chance to get deeper into the deprication warnings then:

```
Accessing type "text" by its string name is deprecated since version 2.8 and will be removed in 3.0. Use the fully-qualified type class name "Symfony\Component\Form\Extension\Core\Type\TextType" instead: 44x
    14x in SeoContentAdminExtensionTest::testExtraPropertyEditView from Symfony\Cmf\Bundle\SeoBundle\Tests\WebTest\Admin
    8x in SeoContentAdminExtensionTest::testItemEditView from Symfony\Cmf\Bundle\SeoBundle\Tests\WebTest\Admin
    8x in SeoContentAdminExtensionTest::testItemCreate from Symfony\Cmf\Bundle\SeoBundle\Tests\WebTest\Admin
    6x in SeoMetadataTypeTest::testDataClassCreationForNonPhpcr
    6x in SeoMetadataTypeTest::testDataClassCreationForPhpcr
    2x in SeoContentAdminExtensionTest::testAdminExtensionExists from Symfony\Cmf\Bundle\SeoBundle\Tests\WebTest\Admin
```
I can't see any issue in seo-bundle in that example.

**Comment by WouterJ** [comment]
You can set the env variable to a regex to match the exception message IIRC. If you check the symfony docs about the phpunit bridge, you can read more info

**Comment by ElectricMaxxx** [comment]
Thx. I will have a look at it tonight.

**Comment by WouterJ** [comment]
Fixed some more things. Most of the todo items in your PR description relate to SonataAdminBundle. At the moment, this PR won't pass as we can't install a Symfony 3 supporting SonataAdminBundle version (see https://github.com/sonata-project/SonataSeoBundle/issues/122)

**Comment by ElectricMaxxx** [comment]
@WouterJ `Defining the initRuntime() method in the "sonata_doctrine_phpcr_admin" extension is deprecated since version 1.23. Use the `needs_environment` option to get the Twig_Environment instance in filters, functions, or tests; or explicitly implement Twig_Extension_InitRuntimeInterface if needed (not recommended): 19x` this one seems to be ours too, or?

**Comment by WouterJ** [comment]
@ElectricMaxxx no, it's of the SonataDoctrinePhpcrAdmin and fixed in 2.0 (2.8 deps still use the 1.x version).

I've a working version locally, just need to fix somethings in the BurgovKeyValueFormBundle and I'll push.

**Comment by ElectricMaxxx** [comment]
Shall we do a PR on burgovKeyValue of is something on it's way?

**Comment by WouterJ** [comment]
Once https://github.com/Burgov/KeyValueFormBundle/pull/23 is merged, the tests should pass.

**Comment by ElectricMaxxx** [comment]
I think they have a parallel and almost equal PR to yours. Will Look for a Link After the current meeting.

**Comment by ElectricMaxxx** [comment]
@WouterJ see https://github.com/Burgov/KeyValueFormBundle/pull/22

**Comment by ElectricMaxxx** [comment]
Both PRs on BurgovKeyValue are still open.

**Comment by WouterJ** [comment]
:tada:

**Pull request merged**